docs/orbs-authoring.md

```diff
@@ -0,0 +1,87 @@
+
+While we do not enforce it strictly, we strongly recommend that when versioning your production orbs you use the standard semver convention for deciding what major, minor, and patch should mean semantically:
+
+MAJOR: when you make incompatible API changes,
+MINOR: when you add functionality in a backwards-compatible manner, and
+PATCH: when you make backwards-compatible bug fixes.
+
+### Using orbs in orbs and register-time resolution
+You may use an `orbs` stanza inside an orb, providing a way to pull in other orbs' elements into your orb. The scoping rules are the same as orbs in configuration, and you declare inline orbs in an orb.
+
+NOTE that because production orb releases are immutable the system will resolve all orb dependencies at the time you register your orb rather than the time you run your build (eager resolution). For example, imagine you go to publish orb `foo/bar` at version `1.2.3`, and that orb you are publishing has, internally, an `orbs` stanza that imports another orb referenced as `biz/baz@volatile`. Let's imagime also that the latest version at the time you want to do this is `2.1.0`. At the time you register `foo/bar@1.2.3` the system will resolve `biz/baz@volatile` as the latest version and include its elements directly into the packaged version of `foo/bar@1.2.3`. Now imagine that afterwards `biz/baz` receives a new release that is now `3.0.0`. Anyone using `foo/bar@1.2.3` will not see the change in `biz/baz#3.0.0` until `foo/bar` is published at a higher version. 
+
+### Yanking production orbs
+
+In general, we prefer to never delete production orbs that were published as Open because it harms reliability and trust of all users in the orb registry as a source of configuration. We recognize there are circumstances in which a production orb ought be deactivated and/or expunged. If you need to yank an orb for emergency reasons please contact CircleCI (NOTE: IF YOU ARE YANKING BECAUSE OF A SECURITY CONCERN PLEASE PRACTICE RESPONSIBLE DISCLOSURE: https://circleci.com/security/).
+
+### Using the CLI to author and publish orbs
+The `circleci` CLI has several handy commands for managing your orb publishing pipeline. The simplest way to learn the CLI is to [install it](https://circleci.com/docs/2.0/local-cli/#installing-the-circleci-public-cli-from-scratch-on-macos-and-linux-distros) and run `circleci help`. Below are some of the most pertinent commands for publishing orbs:
+
+`circleci namespace create NAME VCS-TYPE ORG-NAME [flags]`
+
+`circleci orb create NAMESPACE ORB [flags]`
+
+`circleci orb validate PATH [flags]`
+
+`circleci orb publish dev PATH NAMESPACE ORB LABEL [flags]`
+
+`circleci orb publish promote NAMESPACE ORB LABEL SEGMENT [flags]`
+
+For full reference use the `help` command inside the CLI, or visit [https://circleci-public.github.io/circleci-cli/circleci_orb.html](https://circleci-public.github.io/circleci-cli/circleci_orb.html) (TODO: make this a link to the main docs once that info is there).
+
+
```

Comment:
Would it be worthwhile to have a "Suggested Workflow" section for authoring orbs? It's much more straightforward to develop an inline orb and then move it into the registry than to develop it standalone. If so, `circleci config process` would also be a good recommendation to validate that an inline orb expands as intended.

Reply:
Agree, but I want to put that out as a new PR once we have a better answer for that. For now, this is better than nothing, but we do want to come back and add a better reference flow.